include/pagmo/algorithm.hpp

```diff
@@ -48,7 +48,14 @@ namespace pagmo
  * This algorithm is used to implement the default constructors of meta-algorithms.
  */
 struct null_algorithm {
-    /// Algorithm implementation
+    /// Algorithm evolve method (juice implementation of the algorithm)
+    /**
+     *
+     * Evolves the population for the requested number of generations.
+     *
+     * @param[in] pop population to be evolved
```

Comment:
Would you mind just replacing all ``@param[in]`` with just ``@param``? In my experience the ``[in]`` just adds noise in the doxygen output, and I have never found it useful.